forsys/pmatrix.py

```diff
@@ -30,30 +25,27 @@ def _build_matrix(self) -> Tuple:
         :return: Returns LHS matrix and RHS matrix
         :rtype: Tuple
         """
-        self.rhs_matrix = np.zeros(len(self.big_edges_to_use))
-
-        self.mapping_order = {value: enumid for enumid, value in enumerate(self.frame.cells.keys())}
+        n_edges, n_cells = len(self.big_edges_to_use), len(self.frame.cells)
+        self.mapping_order = {key: enumid for enumid, key in enumerate(self.frame.cells)}
 
+        self.rhs_matrix = np.empty(n_edges)
+        self.lhs_matrix = np.empty((n_edges, n_cells))
         for position_id, big_edge in enumerate(self.big_edges_to_use):
-            lhs_row, rhs_value =  self.get_row(big_edge)
-
+            lhs_row, rhs_value = self.get_row(big_edge)
             self.rhs_matrix[position_id] = rhs_value
-            if not self.lhs_matrix:
-                self.lhs_matrix = Matrix(lhs_row).T
-            else:
-                self.lhs_matrix = self.lhs_matrix.row_insert(self.lhs_matrix.shape[0],
-                                                             Matrix(lhs_row).T)
-        
-        self.removed_columns = []
-        for column in range(self.lhs_matrix.shape[1]):
-            if np.all([val == 0 for val in self.lhs_matrix.col(column)]):
-                self.removed_columns.append(column)
-
-        ii = 0
-        for element in self.removed_columns:
-            self.lhs_matrix.col_del(element - ii)
-            # self.mapping_order["r_"+str(ii)] = element
-            ii += 1
+            self.lhs_matrix[position_id] = lhs_row
+
+        # remove columns with all zeros
+        self.removed_columns = np.nonzero(np.all(self.lhs_matrix == 0, axis=0))[0].tolist()  # list of column indices

```

Comment:
- `mat == 0` gives a boolean mask with Ts where values are 0, and Fs everywhere else
- `np.all(mask, axis=0)` produces a boolean vector of size `mask.shape[1]` with Ts where all values in the column equal one and Fs everywhere else
- np.nonzero is used to get indices where the vector values are T (columns that we want to remove)